keesat/law-reading-robot-data
# Create summarization.py

## What I did

Created a script that takes a long text as input and uses various text processing and language model techniques to generate a concise summary of the input text. The code splits the text into smaller chunks, performs summarization, and returns the resulting summary.


## Issue #s Addressed
https://trello.com/c/6F41dJo4/22-script-for-getting-summaries-of-bills
https://trello.com/c/tNRAPspw/50-look-into-langchain-for-large-text-parsing-and-summarization

- closes #

## How to test it
To test the code:
1. Please ensure to install the required libraries for the code to work correctly.
  `  pip install langchain gpt4all chromadb langchainhub openai`
2. Provide a long text as input to the `get_gpt_summary` function.
3. The function will return the resulting summary and can be accessed as `result["text"]`.
